src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -212,6 +208,40 @@ class MediaBaseViewer extends BaseViewer {
     }
 
     /**
+     * Handler for autoplay
+     *
+     * @private
+     * @emits autoplay
+     * @return {void}
+     */
+    handleAutoplay() {
+        const isAutoplayEnabled =
+            this.cache.has(MEDIA_AUTOPLAY_CACHE_KEY) && this.cache.get(MEDIA_AUTOPLAY_CACHE_KEY) === 'Enabled';
+        this.emit('autoplay', isAutoplayEnabled);
+    }
+
+    /**
+     * Determines if media should autoplay based on cached settings value.
+     *
+     * @private
+     * @emits volume
+     * @return {void}
+     */
+    checkAutoplay() {
+        if (
+            (!this.cache.has(MEDIA_AUTOPLAY_CACHE_KEY) || this.cache.get(MEDIA_AUTOPLAY_CACHE_KEY) === 'Disabled') &&
+            !Browser.isIOS()
+        ) {
+            return;
+        }
+
+        // iOS doesn't fire loadeddata event until some data loads
+        // Adding autoplay prevents that, but iOS Safari won't autoplay media
+        // https://webkit.org/blog/6784/new-video-policies-for-ios/
+        this.mediaEl.autoplay = true;
```

Comment:
Could this be triggered when the media element isn't loaded yet?

Reply:
It should be set before the media element is loaded. From MDN: `A Boolean attribute; if specified, the video automatically begins to play back as soon as it can do so without stopping to finish loading the data.`